### PR TITLE
Blockly Added Capability

### PR DESCRIPTION
**Pull request by AmbientOSX**
...brings back the trashcan, removes some flyout offsets since we're using advanced categories and should fix tooltips and mutator callout origins for Blockly.

@kadst43 @BrettASwift @nmarshak1337

### CODE REVIEW

File: support/client/lib/vwf/view/blockly.js [hunk 1]
```diff
@@ -451,11 +450,13 @@ define( [ "module", "vwf/view", "jquery", "vwf/model/blockly/JS-Interpreter/acor
     function cleanUpdateToolbox( xml ) {
         // negateOverlap is being use to negate some of the effect of the MARGIN variable
         // in Blockly.createDom_ when deleting blocks over the flyout
-        var negateOverlap = 35;
+        // SJF: Overlap not present in Blockly with categories enabled
+
+        // var negateOverlap = 35;
         Blockly.updateToolbox( xml );
```
Comment (n8xm) [inline support/client/lib/vwf/view/blockly.js:456]: If the only thing that's executing in this function is `Blockly.updateToolbox( xml )`, does it make more sense to comment out the entire `cleanUpdateToolbox` function, and replace it with `updateToolbox`?

Comment (AmbientOSX) [inline support/client/lib/vwf/view/blockly.js:456]: Leaving it incase we switch back to using basic blockly with no categories or build our own categories or something. Good point though :)

Comment (kadst43) [inline support/client/lib/vwf/view/blockly.js:456]: Hrm.  Might be better to remove the cruft, and put in a comment saying something like "If you ever want to change back to the simple UI, look in github back to thus-and-such a date/branch."

On a related note, though... are level designers going to have the option to switch between the two UIs, or do we have to pick one and stick with it?

Comment (AmbientOSX) [inline support/client/lib/vwf/view/blockly.js:456]: Unfortunately you need to pick one and stick with it for the whole "game" - once it loads it must stay in a certain configuration. Haven't explored what reloading might mean/entail but I suspect it isn't pretty.

**Pull request merged**